### ISSUE TITLE
Jsdoc `@template` issue in syntactic classifier

### ISSUE THREAD

**Issue by basarat** [Microsoft/TypeScript#8676]
Consider the following: 

```ts
/**
 * @template T
 */
function ident<T>(x:T):T {
    return ident;
}
```

`getEncodedSyntacticClassifications` will fail with `Cannot read property 'text' of undefined`

**Stack**:
```
NodeObject.createChildren	@	ntypescript.js:50808
NodeObject.getChildren	@	ntypescript.js:50845
processJSDocTemplateTag	@	ntypescript.js:56304
classifyJSDocComment	@	ntypescript.js:56270
classifyComment	@	ntypescript.js:56243
classifyLeadingTriviaAndGetTokenStart	@	ntypescript.js:56213
tryClassifyNode	@	ntypescript.js:56345
processElement	@	ntypescript.js:56482
processElement	@	ntypescript.js:56484
processElement	@	ntypescript.js:56484
getEncodedSyntacticClassifications	@	ntypescript.js:56185
```

**Reason**: `sourceFile` is `undefined` somewhere in the stack

**Version** : latest as of this morning

refs https://github.com/alm-tools/alm/issues/82 :rose:

**Comment by basarat** [comment]
Didn't want to pollute the issue with my potentially wrong analysis. This screenshot might help people figure out the fix

![image](https://cloud.githubusercontent.com/assets/874898/15379025/06a70fb4-1dac-11e6-8403-9f0dea41c4ee.png)

`getChildren` calls `createChildren` (both without the optional `sourceFile`) and since parent is not there it crashes .... I haven't taken a deeper look :rose:

**Comment by mhegazy** [comment]
@zhengbli was looking into a similar issue the other day.

**Comment by zhengbli** [comment]
@basarat can you give https://github.com/Microsoft/TypeScript/pull/8702 a try? Thanks!

**Comment by mhegazy** [comment]
@basarat can you give https://github.com/Microsoft/TypeScript/pull/8103 a try instead?

**Comment by basarat** [comment]
I just made a new ntypescript build https://github.com/TypeStrong/ntypescript that has that PR https://github.com/Microsoft/TypeScript/tree/166f399d176486a1247c54dcd6ad4a18e33c18f3 but still no dice:

![image](https://cloud.githubusercontent.com/assets/874898/15697208/6f494196-27fb-11e6-8b0c-4f04d875f1a7.png)

Thanks for pinging / attempting :rose:

**Comment by basarat** [comment]
I'm also getting a weird error in the latest build where a property is defined (e.g. `styles` below) but `this.styles` is showing an error.

![image](https://cloud.githubusercontent.com/assets/874898/15697399/101e12da-27fd-11e6-8b04-003c61721a4d.png)

But haven't been able to come up with a short repro ... so you can ignore that for now :rose:

**Comment by zhengbli** [comment]
It turns out when calling `parseIsolatedJSDocComment`, the fixing node's parent pointer part was skipped. This should be fixed by #8930, also added a test just for sure.

**Comment by basarat** [comment]
Verified. Thanks :100: :rose:
